copy.js

```diff
@@ -28,6 +28,19 @@ function maybe_clean (orig_url, urls, f) {
     return (res != '' ? res : orig_url);
 }
 
+// clean URL
+function maybe_clean_url (orig_url, urls, f) {
+    var res = '';
+    for (var idx = 0; idx < urls.length; idx++) {
+        var patt = glob_to_regexp(urls[idx]);
+        if (patt.test(orig_url)) {
+            res = f(orig_url)['redirectUrl'];
+            break;
+        }
+    }
+    return (res != '' ? res : orig_url);
```

Comment:
oh yeah, and this adds amazon links cleaning: this function can be used to clean more URL (f.e. maybe a "de-ampifier" too?). I've duplicated the code of `maybe_clean` (which take care of query params) and that's not cool, but for the moment I would keep it like that, let's see first which direction `maybe_clean_url` will take.

Reply:
btw my naming choices are so bad :^)